server/models/recipe.js

```diff
@@ -0,0 +1,51 @@
+import mongoose from 'mongoose'
+import slugs from 'slugs'
+
+const recipeSchema = new mongoose.Schema({
+
+  title: {
+    type: String,
+    required: true
+  },
+  slug: String,
+  image: {
+    type: String,
+    required: true
+  },
+  author: {
+    type: String,
+    required: true
+  },
+  description: {
+    type: String,
+    required: true
+  },
+  ingredients: {
+    type: [String],
+    required: false
+  },
+  steps: {
+    type: [String],
+    required: false
+  },
+  category: {
+    type: String,
+    required: true
+  },
+  likes: {
+    type: Number,
+    required: false
+  }
+})
+
+recipeSchema.pre('save', function(next){
+  if (!this.isModified('title')) next()
+
+  this.slug = slugs(this.title)
+
+  next()
+})
```

Comment:
the goal of this function is that it only runs the slugs function on the recipe title when the title is modified so when user edit something else it won't run the slugs function but it should save the recipe because i am calling next()

however i think this code will run the slugs function everytime even if the title is not modified because i forgot to add a return statement inside the if blocks!

recipeSchema.pre('save', function(next){
 +  if (!this.isModified('title')) {

next()
return;
}
 +
 +  this.slug = slugs(this.title)
 +
 +  next()
 +})